src/lib/annotations/drawing/DrawingPath.js

```diff
@@ -18,21 +23,39 @@ class DrawingPath {
     minY = Infinity;
 
     /**
+     * [constructor]
+     *
+     * @param {Object} drawingPathData - The drawingPath object data to be instantiated into an object
+     * @return {DrawingPath} A DrawingPath instance
+     */
+    constructor(drawingPathData) {
+        if (drawingPathData) {
+            this.path = drawingPathData.path.map((num) =>
+                annotatorUtil.createLocation(parseFloat(num.x), parseFloat(num.y))
+            );
+            this.maxX = drawingPathData.maxX;
+            this.minX = drawingPathData.minX;
+            this.maxY = drawingPathData.maxY;
+            this.minY = drawingPathData.minY;
+        }
+    }
+
+    /**
      * Add position to coordinates and update the bounding box
      *
-     * @param {number} xPos - xPosition to be part of the drawing
-     * @param {number} yPos - yPosition to be part of the drawing
+     * @param {Location} documentLocation - Original document location coordinate to be part of the drawing path
+     * @param {Location} [browserLocation] - Optional browser position to be saved to browserPath
      * @return {void}
      */
-    addCoordinate(xPos, yPos) {
-        if (!xPos || !yPos) {
+    addCoordinate(documentLocation, browserLocation) {
+        if (!documentLocation || !documentLocation.x || !documentLocation.y) {
             return;
         }
 
         // OPTIMIZE (@minhnguyen): We convert a number to a string using toFixed and then back a number.
```

Comment:
Unfortunately not, the native function to truncate is number.toFixed(precision) which provides a string.